src/find_meaningful_words.py

```diff
@@ -0,0 +1,143 @@
+                                        for x in range_base_words]
+    # Create a words_ count tuple list
+    base_words_relative_reference_count_tuple_list = [
+        (base_words_count_tuple_list[x][WORDS_PLACE_IN_FREQUENCY_LIST], 
+        base_relative_to_reference_count[x]) 
+        for x in range_base_words]
+    return base_words_relative_reference_count_tuple_list
+
+
+def get_ratio_of_base_words_count_relative_to_reference_words_count(base_words_count_tuple_list,
+                                                                    reference_words_count_tuple_list):
+    base_words_list = turn_words_count_tuple_list_to_words_only(
+        base_words_count_tuple_list)
+    reference_words_list = turn_words_count_tuple_list_to_words_only(
+        reference_words_count_tuple_list)
+    base_words_with_reference_count = get_base_words_with_reference_count(
+        base_words_list, reference_words_list, reference_words_count_tuple_list)
+    base_words_relative_reference_count_tuple_list = calculate_relative_words_count_of_2_lists(
+        base_words_count_tuple_list, base_words_with_reference_count)
+
+    return base_words_relative_reference_count_tuple_list
+
+
+def find_meaningful_words_in_file_from_another_file(base_path, reference_path, number_of_words):
+    # Count words in base and reference files
+    base_words_counted = count_words_in_list(
+        get_words_only_small_letters(SaveRead.read_txt_file_to_string(base_path)))
+    reference_words_counted = count_words_in_list(
+        get_words_only_small_letters(SaveRead.read_txt_file_to_string(reference_path)))
+    # Get the ratio of base words count relative to reference words count.
+    base_words_relative_reference_count_tuple_list = get_ratio_of_base_words_count_relative_to_reference_words_count(
+        base_words_counted, reference_words_counted)
+    # Sort the relative count of base words
+    sorted_base_words_relative_reference_count_tuple_list = sort_words_count(
+        base_words_relative_reference_count_tuple_list)
+    most_meaningful_words = get_only_common_words(
+        sorted_base_words_relative_reference_count_tuple_list, number_of_words)
+    return most_meaningful_words
+
+
+def main(print_results=True):
+    hamlet_path = "hamlet.txt"
+    # Find how many words are there in hamlet txt file.
+    hamlet_words_number = count_words_from_file(hamlet_path)
+    if print_results:
+        print(f"There are {hamlet_words_number} words in hamlet play.")
+
+    # Find the 30 most common words.
+    hamlet_path = "hamlet.txt"
+    how_many_common_words = 30
+    hamlet_most_common_words = get_most_common_words_in_string(
+        hamlet_path, how_many_common_words)
+    if print_results:
+        print(
+            f"The {how_many_common_words} most common words in hamlet are - {hamlet_most_common_words}")
+
+    # Having macbeth txt, could you find the most meaningful words in the hamlet file.
+    hamlet_path = "hamlet.txt"
```

Comment:
why is hamlet_path redefined?

Reply:
I think it's a matter of preference, it's redefined to help the code maintainer understand that it's being used again. 
Maybe no need?

Reply:
What do you think, would you do it?

I've added it after I saw that when you looked at the third function and thought I'm not using the Hamlet file but I'm using the Macbeth file instead. So I redefined it to help have better clarity.